### PR TITLE
[Fix - ADF 90] Advanced Search: Empty criteria drop-down list at the first attempt to open it

### PR DESCRIPTION
**Pull request by habemuscode**
**Issue**

https://oat-sa.atlassian.net/browse/ADF-90

**PR needed before merge this**

https://github.com/oat-sa/tao-core-ui-fe/pull/254

**Description**

In the modal when the user opens the Criteria list for the first time, it's empty. When the Criteria list is opened the next time - the list of existed properties appears. Reproduces every time the modal opened.

**Steps to reproduce**

- Trigger a search modal from the "Search bar";
- Click "add criteria".

**Actual result**

No items shown in the drop-down list.

**Expected result**

Existing properties are shown in the criteria drop-down list after the first attempt to open it. If the operation is done asynchronous and takes a bit longer (but no longer than a few seconds), a loading indicator replaces the ‘+' icon in the link until the drop-down is fully populated, to provide feedback for the user that an operation is in progress. The drop-down only appears when it is fully populated, only then, the '+' icon replaces the loading indicator.

### CODE REVIEW

**Comment by habemuscode** [comment]
Deployed temporarily here: http://test-as-acl-fix-with-elastic.playground.kitchen.it.taocloud.org:47061

File: views/package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
         "@oat-sa/tao-core-libs": "0.4.3",
         "@oat-sa/tao-core-sdk": "^1.10.1",
         "@oat-sa/tao-core-shared-libs": "1.0.3",
-        "@oat-sa/tao-core-ui": "1.22.10",
+        "@oat-sa/tao-core-ui": "git://github.com/oat-sa/tao-core-ui-fe#fix/ADF-90/loading-icon-while-metadata-request",
```
Comment (mike-ionut-mihai-sandu-tao) [inline views/package.json:16]: if you want to merge this to develop then i think the PR is already failed :)

Comment (habemuscode) [reply views/package.json:16]: This is just for deploy to Playground to be able to engineers and QA, test before merge. When I get approvals, then I will put the correct version before merge

**Review by gabrielfs7 — APPROVED** [review]
LGTM, just do not forget to update packages.json before merge :) 

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Review by voknil — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [ ]  New code is not subject to concurrency issues (if applicable)
- [ ]  Feature is working correctly on my local machine (if applicable)
- [ ]  Acceptance criteria are respected
- [ ]  Pull request title and description are meaningful

**Pull request merged**